File: ios/Sources/VaultSync/Domain/Repositories/FileStorage.swift
import Foundation

public protocol FileStorage: Sendable {
    func readFile(at path: String) async throws -> Data
    func writeFile(at path: String, data: Data) async throws -> String
    func deleteFile(at path: String) async throws -> Bool
    func fileExists(at path: String) async -> Bool
    func getFileSize(at path: String) async throws -> Int64
    func copyFile(from sourcePath: String, to destinationPath: String) async throws -> String
}

File: android/app/src/main/java/com/vaultsync/domain/repository/FileStorage.kt
package com.vaultsync.domain.repository

interface FileStorage {
    suspend fun readFile(path: String): ByteArray
    suspend fun writeFile(path: String, data: ByteArray): String
    suspend fun deleteFile(path: String): Boolean
    suspend fun fileExists(path: String): Boolean
    suspend fun getFileSize(path: String): Long
    suspend fun copyFile(sourcePath: String, destinationPath: String): String
}
